tests/tools/test_features_ranking.py

```diff
@@ -0,0 +1,277 @@
+
+        result_bf = _utils._adjust_pvalues(pvals, corr_method="bonferroni")
+
+        for group in groups:
+            assert (pvals[group] <= result_bf[group]).all()
+            assert np.allclose(result_bf[group], expected_result_bf[group])
+
+    def test_sort_features(self):
+        def _is_sorted(arr: np.array):
+            return np.all(arr[:-1] <= arr[1:])
+
+        adata = ep.dt.mimic_2(encoded=False)
+        adata.uns["rank_features_groups"] = {
+            "params": {"whatever": "here is", "it does": "not matter", "but this key should be": "present for testing"},
+            "names": pd.DataFrame(
+                {"group1": ("gene2", "gene1", "gene4", "gene3"), "group2": ("gene5", "gene6", "gene7", "gene8")}
+            ).to_records(index=False),
+            # Let's mix genes in group1, and leave them sorted in group2
+            "pvals": pd.DataFrame({"group1": (0.02, 0.01, 1.00, 0.03), "group2": (0.04, 0.05, 0.06, 0.99)}).to_records(
+                index=False
+            ),
+            "scores": pd.DataFrame({"group1": (2, 1, 100, 3), "group2": (4, 5, 6, 7)}).to_records(index=False),
+            "log2foldchanges": pd.DataFrame({"group1": (2, 1, 10, 3), "group2": (4, 5, 6, 7)}).to_records(index=False),
+        }
+        # Doesn't really matter that they are the same here but order should be preserved
+        adata.uns["rank_features_groups"]["pvals_adj"] = adata.uns["rank_features_groups"]["pvals"].copy()
+
+        _utils._sort_features(adata)
+
+        # Check that every feature is sorted
+        # Actually, some of them would be sorted in the opposite direction (e.g. scores) for real data
+        # But what matters here is that the permutation is the same for every key and is based on adjusted p-values
+        for key in adata.uns["rank_features_groups"]:
+            if key == "params":
+                continue
+            assert _is_sorted(adata.uns["rank_features_groups"][key]["group1"])
+            assert _is_sorted(adata.uns["rank_features_groups"][key]["group2"])
+
+    def test_save_rank_features_result(self):
+        groups = ("group1", "group2")
+
+        adata = ep.dt.mimic_2(encoded=False)
+        adata.uns["rank_features_groups"] = {
+            "params": {"whatever": "here is", "it does": "not matter", "but this key should be": "present for testing"}
+        }
+
+        names = pd.DataFrame(
+            {"group1": ("gene2", "gene1", "gene4", "gene3"), "group2": ("gene5", "gene6", "gene7", "gene8")}
+        ).to_records(index=False)
+        pvals = pd.DataFrame({"group1": (0.02, 0.01, 1.00, 0.03), "group2": (0.04, 0.05, 0.06, 0.99)}).to_records(
+            index=False
+        )
+        scores = pd.DataFrame({"group1": (2, 1, 100, 3), "group2": (4, 5, 6, 7)}).to_records(index=False)
+        logfoldchanges = pd.DataFrame({"group1": (2, 1, 10, 3), "group2": (4, 5, 6, 7)}).to_records(index=False)
+
+        # Chack that adding onle required keys works
```

Comment:
```suggestion
        # Check that adding only required keys works
```

Reply:
We all have that inner cat who sometimes randomly presses keyboard buttons

Reply:
Mine is apparently drunk